From cd4450687ae075cb15d42c87e7d5a3a1ab764079 Mon Sep 17 00:00:00 2001
From: ADX12 upstream sync <adx12-sync@invalid.example>
Date: Wed, 26 Aug 2026 19:12:42 +0530
Subject: [PATCH] d3d12: use UAV texture storage for float clears

---
 src/d3d12/d3d12_command_list.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/d3d12/d3d12_command_list.cpp b/src/d3d12/d3d12_command_list.cpp
index 3b599ae..8816b5c 100644
--- a/src/d3d12/d3d12_command_list.cpp
+++ b/src/d3d12/d3d12_command_list.cpp
@@ -1441,10 +1441,10 @@ public:
       break;
     }
     case ShaderVisibleDescriptorType::UAVTexture: {
-      allocator_->clear_uav_.begin(color, Descriptor.SRVTexture.texture, Descriptor.SRVTexture.view);
+      allocator_->clear_uav_.begin(color, Descriptor.UAVTexture.texture, Descriptor.UAVTexture.view);
       full_rect = {
-          0, 0, (LONG)Descriptor.SRVTexture.texture->width(Descriptor.SRVTexture.view),
-          (LONG)Descriptor.SRVTexture.texture->height(Descriptor.SRVTexture.view)
+          0, 0, (LONG)Descriptor.UAVTexture.texture->width(Descriptor.UAVTexture.view),
+          (LONG)Descriptor.UAVTexture.texture->height(Descriptor.UAVTexture.view)
       };
       break;
     }
-- 
2.50.1

